test/end2end_test.go

```
@@ -3585,6 +3586,162 @@ func testClientStreamingError(t *testing.T, e env) {
+	defer cancel()
+
+	stream, err := ss.Client.StreamingInputCall(ctx)
+	if err != nil {
+		t.Fatalf(".StreamingInputCall(_) = _, %v, want <nil>", err)
+	}
+
+	_, err = stream.CloseAndRecv()
+	if err == nil {
+		t.Fatalf("%v.CloseAndRecv() = %v, want an error", stream, err)
+	}
+	if status.Code(err) != codes.Internal {
+		t.Fatalf("%v.CloseAndRecv() = %v, want error %s", stream, err, codes.Internal)
+	}
+}
+
+// Tests the scenario where a client-streaming server calls `Recv()` after calling
+// `SendAndClose()`. And error is expected if `Recv()` is called after calling
+// SendAndClose()` i.e the RPC should be terminated.
+func (s) TestClientStreamingCardinalityViolation_ServerHandlerRecvAfterSendAndClose(t *testing.T) {
+	// TODO : https://github.com/grpc/grpc-go/issues/8119 - remove `t.Skip()`
+	// after this is fixed.
+	t.Skip()
+	ss := &stubserver.StubServer{
+		StreamingInputCallF: func(stream testgrpc.TestService_StreamingInputCallServer) error {
+			sum := 0
+			in, _ := stream.Recv()
+			p := in.GetPayload().GetBody()
+			sum += len(p)
+			stream.SendAndClose(&testpb.StreamingInputCallResponse{
+				AggregatedPayloadSize: int32(sum),
+			})
+			_, err := stream.Recv()
+			return err
+		},
+	}
+	if err := ss.Start(nil); err != nil {
+		t.Fatalf("Error starting endpoint server: %v", err)
+	}
+	defer ss.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	stream, err := ss.Client.StreamingInputCall(ctx)
+	if err != nil {
+		t.Fatalf("StreamingInputCall(_) = _, %v, want <nil>", err)
+	}
+	payload, err := newPayload(testpb.PayloadType_COMPRESSABLE, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	req := &testpb.StreamingInputCallRequest{
+		Payload: payload,
+	}
+
+	for {
+		if err = stream.Send(req); err == nil {
+			continue
+		}
+		select {
+		case <-ctx.Done():
+			t.Fatal("timed out waiting for error from server")
+		default:
+		}
+		break
+	}
+	_, err = stream.CloseAndRecv()
+	if err == nil {
```

same here. Isn't client expected to ignore the error?

---

Not sure if it was missed or not clarified. If you do as mentioned above, the client is not expected to receive any error. So, you should check for err != nil and terminate with failure if that's the case.

---

done.